include/revng/Model/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+# Header files containing model YAML must be added here
+set(MODEL_HEADERS
+    CABIFunctionType.h
+    EnumType.h
+    PrimitiveType.h
+    PrimitiveTypeKind.h
+    RawFunctionType.h
+    StructType.h
+    TypedefType.h
+    Type.h
+    TypeKind.h
+    UnionType.h
+    ABI.h
+    AbstractFunctionEdge.h
+    Architecture.h
+    Argument.h
+    BasicBlock.h
+    Binary.h
+    CallEdge.h
+    DynamicFunction.h
+    EnumEntry.h
+    FunctionEdge.h
+    FunctionEdgeType.h
+    Function.h
+    FunctionType.h
+    NamedTypedRegister.h
+    QualifiedType.h
+    Qualifier.h
+    QualifierKind.h
+    Register.h
+    RegisterState.h
+    Segment.h
+    StructField.h
+    TypedRegister.h
+    UnionField.h
+)
+
+set(MODEL_TEMPLATES
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/class_forward_decls.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/enum.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_forward_decls.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_late.tpl.hpp"
+    "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/tuple_tree_generator/templates/struct_impl.tpl.cpp"
+)
+
+# -- Collect YAML from manually written headers
+set(COLLECTED_YAML_PATH "${CMAKE_BINARY_DIR}/include/revng/Model/model_schema.yml")
+
+add_custom_command(
+    OUTPUT "${COLLECTED_YAML_PATH}"
+    COMMAND "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/extract_yaml.py"
+    ARGS "MODEL-YAML" ${MODEL_HEADERS} > "${COLLECTED_YAML_PATH}"
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+    DEPENDS ${MODEL_HEADERS}
+)
+
+# -- Generate C++ headers from collected yaml
+set(OUTPUT_DIR "${CMAKE_BINARY_DIR}/include/revng/Model/Generated")
+
+foreach(HEADER ${MODEL_HEADERS})
+  get_filename_component(HEADER_FILENAME "${HEADER}" NAME)
+  get_filename_component(HEADER_FILENAME_WE "${HEADER}" NAME_WE)
+  set(EARLY_OUTPUT "${OUTPUT_DIR}/Early/${HEADER_FILENAME}")
+  set(LATE_OUTPUT "${OUTPUT_DIR}/Late/${HEADER_FILENAME}")
+  set(IMPL_OUTPUT "${OUTPUT_DIR}/Impl/${HEADER_FILENAME_WE}.cpp")
+  list(APPEND GENERATED_HEADERS "${EARLY_OUTPUT}")
+  list(APPEND GENERATED_HEADERS "${LATE_OUTPUT}")
+  list(APPEND GENERATED_IMPLS "${IMPL_OUTPUT}")
+endforeach()
+list(APPEND GENERATED_HEADERS "${OUTPUT_DIR}/ForwardDecls.h")
+
+add_custom_command(
+    COMMAND "${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator/main.py"
+    ARGS --cpp-headers "${COLLECTED_YAML_PATH}" "${OUTPUT_DIR}"
```

Comment:
Would `python ${CMAKE_SOURCE_DIR}/scripts/tuple_tree_generator` work?

Reply:
I have to rework the package a bit (to turn `main.py` into `__main__.py`) but yeah, it can work

Reply:
Scratch that, we would need to add the module to PYTHONPATH, which is even uglier. I'd leave it the way it is.